packages/itwinui-css/src/input-container/input-container.scss

```diff
@@ -67,17 +47,7 @@
       }
 
       .iui-select-button {
-        border-block-end: transparent;
-
-        &::after {
-          content: '';
-          inline-size: 100%;
-          position: absolute;
-          inset-block-end: 0;
-          inset-inline-start: 0;
-          block-size: var(--iui-size-3xs);
-          background-color: var(--iui-color-border-#{$status});
-        }
+        border-block-end: var(--iui-size-3xs) solid var(--iui-color-border-#{$status});
       }
```

Comment:
I realized that adding `border-block-end` similar to `iui-input` ([code](https://github.com/iTwin/iTwinUI/pull/1753/files#diff-79a618b1da06b9cedff0df3a138d8a8da22a8185e8f0b5a2cf6c8aeb23a49b5cL51)) also gave the same effect as the `&::after {…}`. So I replaced the multi-line `&::after {…}` with a single line `border-block-end`.

Reply:
I believe the reason for using `::after` was to preserve vertical text alignment. If the bottom border is 2px and the top border is 1px, then the text does not appear centered.

Reply:
Actually I'm no longer sure what "select with status" should look like. The old demos looked like this:
![image](https://github.com/iTwin/iTwinUI/assets/9084735/5356b6a7-dfb8-4e36-a381-95cf8b39a645)

I'm trying to investigate.

Reply:
**Update**: removed `::after` in [`3deac83` (#1753)](https://github.com/iTwin/iTwinUI/pull/1753/commits/3deac837b3a5da4b032da5b446ab3d2ddb09e5e4) since it was doing nothing

the vertical centering is handled using `padding-block-end`